src/tdyio.c

```diff
@@ -122,24 +138,36 @@ PetscErrorCode TdyIOInitializeHDF5(char *ofilename, DM dm){
   PetscFunctionReturn(0);
 }
 
-PetscErrorCode TdyIOWriteHDF5Var(char *ofilename, Vec U,PetscReal time){   
+PetscErrorCode TdyIOWriteHDF5Var(char *ofilename,DM dm,Vec P,Vec S,PetscReal time){   
   PetscViewer viewer;
   PetscErrorCode ierr;
   PetscInt numCell;
   char word[32];
   PetscMPIInt rank;
+  PetscSection sec;
 
   ierr = MPI_Comm_rank(PETSC_COMM_WORLD,&rank); CHKERRQ(ierr);
   
   sprintf(word,"%11.5e",time);
+  
   ierr = PetscViewerHDF5Open(PETSC_COMM_WORLD,ofilename,FILE_MODE_APPEND,&viewer);CHKERRQ(ierr);
-  ierr = PetscObjectSetName((PetscObject) U,word);CHKERRQ(ierr);
-  ierr = VecView(U,viewer);CHKERRQ(ierr);  
+
+  ierr = PetscObjectSetName((PetscObject) P,word);CHKERRQ(ierr);
+  ierr = VecView(P,viewer);CHKERRQ(ierr);
+
+  ierr = PetscObjectSetName((PetscObject) S,word);CHKERRQ(ierr);
+  ierr = DMGetSection(dm, &sec);
+  //Change to field name to Saturation
+  ierr = PetscSectionSetFieldName(sec, 0, "Saturation"); CHKERRQ(ierr);
+  ierr = VecView(S,viewer);CHKERRQ(ierr);
+  //Change back to LiquidPressure
+  ierr = PetscSectionSetFieldName(sec, 0, "LiquidPressure"); CHKERRQ(ierr);
```

Comment:
Why is this switching taking place?

cc: @knepley

Reply:
We can temporarily change the name of the field in the section so it prints out correctly with VecView() and then change it back after the call to VecView()

I'm not sure if there is a more elegant way to do this though

Reply:
Ok. Let's change the function such that we only write one variable at a time with the variable name as an input argument (see [psuedo code](https://github.com/TDycores-Project/TDycore/pull/179/files#r633853680)). This would enable us to just set the fieldname using the variable name.